src/Controllers/BohrungController.cs

```diff
@@ -10,22 +11,98 @@ namespace EWS;
 [Route("[controller]")]
 public class BohrungController : EwsControllerBase<Bohrung>
 {
-    public BohrungController(EwsContext context)
+    private readonly HttpClient client;
+    private readonly ILogger<DataServiceController> logger;
+    private readonly EwsContext context;
+
+    public BohrungController(HttpClient client, ILogger<DataServiceController> logger, EwsContext context)
         : base(context)
     {
+        this.client = client;
+        this.logger = logger;
+        this.context = context;
     }
 
     /// <inheritdoc/>
-    public override Task<IActionResult> CreateAsync(Bohrung entity)
+    public override async Task<IActionResult> CreateAsync(Bohrung entity)
     {
         entity.Bohrprofile = null;
-        return base.CreateAsync(entity);
+        if (entity.Geometrie == null)
+        {
+            return await base.CreateAsync(entity).ConfigureAwait(false);
+        }
+        else
+        {
+            return await UpdateStandortBeforeContinouing(base.CreateAsync, entity).ConfigureAwait(false);
+        }
     }
 
     /// <inheritdoc/>
-    public override Task<IActionResult> EditAsync(Bohrung entity)
+    public override async Task<IActionResult> EditAsync(Bohrung entity)
     {
         entity.Bohrprofile = null;
-        return base.EditAsync(entity);
+        if (entity.Geometrie == null)
+        {
+            return await base.EditAsync(entity).ConfigureAwait(false);
+        }
+        else
+        {
+            return await UpdateStandortBeforeContinouing(base.EditAsync, entity).ConfigureAwait(false);
+        }
+    }
+
+    private async Task<IActionResult> UpdateStandortBeforeContinouing(Func<Bohrung, Task<IActionResult>> operation, Bohrung item)
```

Comment:
PP: Aussagekräftigeren Namen verwenden, z.B.
```suggestion
    private async Task<IActionResult> UpdateStandortAndBohrung(Func<Bohrung, Task<IActionResult>> createOrUpdateBohrung, Bohrung item)
```